connector/kinesis-asl/pom.xml

```diff
@@ -54,14 +54,38 @@
       <scope>test</scope>
     </dependency>
     <dependency>
-      <groupId>com.amazonaws</groupId>
+      <groupId>software.amazon.kinesis</groupId>
       <artifactId>amazon-kinesis-client</artifactId>
```

Comment:
Hi @pan3793, KCL (software.amazon.kinesis:amazon-kinesis-client) is built on top of Kinesis Data Stream API (software.amazon.awssdk:kinesis), and KCL provides additional functionalities such as load balancing, error recovery etc. According to this doc, generally it is recommended to use KCL over Kinesis Data Stream API:https://docs.aws.amazon.com/streams/latest/dev/developing-consumers-with-sdk.html .  

Perhaps we could try to raise an issue to the [KCL repo](https://github.com/awslabs/amazon-kinesis-client) and ask if they can release a "clean" version of the library with all the third-party dependencies shaded. But currently I'm not aware there is such a hygienic version of KCL. I guess we probably have to live with this issue if we use KCL.

Reply:
> Perhaps we could try to raise an issue to the [KCL repo](https://github.com/awslabs/amazon-kinesis-client) and ask if they can release a "clean" version of the library with all the third-party dependencies shaded.

Would you like to drive this? It will benefit all downstream projects.

Reply:
Hi @pan3793, I submitted an ticket to the KCL repo: https://github.com/awslabs/amazon-kinesis-client/issues/1245. Let's see what response we would get. On the other hand, it seems to me that this would be a follow-up task, rather than a blocker for this PR? Please correct me if I am wrong.